cloudify_azure/resources/deployment.py

```diff
@@ -94,16 +91,19 @@ def format_params(params):
         def convert_value(v):
             if v is None:
                 return None
-            if isinstance(v, int) or isinstance(v, bool) or isinstance(v, str):
+            if isinstance(v, int) or \
+                    isinstance(v, bool) or \
+                    isinstance(v, text_type):
                 return v
-            if isinstance(v, unicode):
+            if (PY2 and isinstance(v, unicode)) or \
+                    isinstance(v, (text_type, str)):
                 return ast.literal_eval(repr(v))
```

Comment:
I am not sure what/why are we trying to return here. I did a little experiment (Polish text makes no sense but uses some non-ASCII characters):

```
➜  ~ python2
Python 2.7.17 (default, Mar 21 2020, 00:47:07) 
[GCC 9.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> u = u'Zażółć gęślą jaźń'
>>> u
u'Za\u017c\xf3\u0142\u0107 g\u0119\u015bl\u0105 ja\u017a\u0144'
>>> ast.literal_eval(repr(u))
u'Za\u017c\xf3\u0142\u0107 g\u0119\u015bl\u0105 ja\u017a\u0144'
```
```
➜  ~ python3
Python 3.8.2 (default, Feb 26 2020, 22:21:03) 
[GCC 9.2.1 20200130] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> s = 'Zażółć gęślą jaźń'
>>> s
'Zażółć gęślą jaźń'
>>> ast.literal_eval(repr(s))
'Zażółć gęślą jaźń'
```

So the output is different for Py2 and Py3.  Is it O.K.?